pyblish/util.py

```diff
@@ -73,9 +117,15 @@ def publish(context=None, plugins=None, targets=None):
     }
 
     # Second pass, the remainder
+    tasks = list(logic.Iterator(plugins, context, state))
     for Plugin, instance in logic.Iterator(plugins, context, state):
         try:
             result = plugin.process(Plugin, context, instance)
+            tasks_processed_count += 1
+            result["percentage"] = (
+                float(tasks_processed_count) / (len(tasks) + len(collectors))
+            )
+            yield result
```

Comment:
Better yield at the very end of the loop. If you look below, you can see how unexpected exceptions are handled. At the moment, it would yield what is supposed to be a proper result before reaching that check, and yield an exception.

Reply:
Good point.. Done :)